plugins/org.jboss.tools.quarkus.ui/src/org/jboss/tools/quarkus/ui/wizard/CodeProjectWizardController.java

```diff
@@ -0,0 +1,213 @@
+package org.jboss.tools.quarkus.ui.wizard;
+
+import static org.jboss.tools.quarkus.ui.QuarkusUIPlugin.PLUGIN_ID;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.buildship.core.BuildConfiguration;
+import org.eclipse.buildship.core.GradleCore;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.m2e.core.embedder.MavenModelManager;
+import org.eclipse.m2e.core.internal.MavenPluginActivator;
+import org.eclipse.m2e.core.project.IProjectConfigurationManager;
+import org.eclipse.m2e.core.project.LocalProjectScanner;
+import org.eclipse.m2e.core.project.MavenProjectInfo;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.jboss.tools.quarkus.core.code.model.QuarkusModelRegistry;
+import org.jboss.tools.quarkus.ui.QuarkusUIPlugin;
+
+public class CodeProjectWizardController {
+
+    private CodeProjectModel model;
+
+    public CodeProjectWizardController(CodeProjectModel model) {
+        this.model = model;
+    }
+    
+    public IStatus run(IProgressMonitor monitor) {
+        IStatus status = getZip(monitor);
+        if (!monitor.isCanceled() 
+                && status.isOK()) {
+            status = createProject(monitor);
+        }
+        return status;
+    }
+
+    private IStatus getZip(IProgressMonitor monitor) {
+        QuarkusModelRegistry manager = QuarkusModelRegistry.getDefault();
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        IStatus status = null;
+        try {
+            status =  manager.zip(model.getEndpoint(),
+                    model.getTool(),
+                    model.getGroupId(), 
+                    model.getArtifactId(), 
+                    model.getVersion(),
+                    model.getClassName(),
+                    model.getPath(),
+                    model.getSelectedExtensions(),
+                    output, monitor);
+            if (!monitor.isCanceled() 
+                    && status.isOK()) {
+                status = unzip(output.toByteArray(), model.getLocation());
+            }
+        } finally {
+            //NOCODE
```

Comment:
the try/finally block could be move around the unzip method only, not sure it has sense around the zip method.
reading this portion also can be confusing because status can be null for some reason and there is no NPE test above.